src/library_wasmfs.js

```diff
@@ -220,7 +220,17 @@ FS.createPreloadedFile = FS_createPreloadedFile;
         return __wasmfs_symlink(targetBuffer, linkpathBuffer);
       });
     },
-    // TODO: readlink
+    readlink: (path) => {
+      var buf = _malloc(4096);
```

Comment:
I was looking through `struct_info.json` but didn't see `PATH_MAX` exactly. I also didn't see any defines from `limits.h`, so I wasn't sure if there was a reason to avoid including it. Now that I'm looking at it, is there a different value that could be used instead of allocating the max each time?

Reply:
I'm not aware of any reason not to add limits.h to struct_info.json. Maybe it just wasn't needed before now? I think it would be good to add, though. Unfortunately `PATH_MAX` is the only safe value for these allocations since there's no way to know ahead of time how long the path will be.

Did we previously decide that 4kb was too large for a stack allocation? If not, we could get rid of the `malloc` here.

Reply:
Ok sounds good, I'll make the changes to struct_info.json. 

I think there was a TODO comment in `js_api.cpp` that `// TODO: PATH_MAX is 4K atm, so it might be good to reduce this somehow.`, so I wasn't sure 4 kb was too large or something.

Reply:
Just curious, is there a reason this approach is preferred over allocating memory in JS? Is it more efficient?

Reply:
Mostly just the extra code code of exporting malloc and free.